gensrc/proto/binlog.proto

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto2";
+
+package starrocks;
+
+import "olap_file.proto";
+import "types.proto";
+
+message RowsetIdPB {
+  optional int64 hi = 1;
+  optional int64 mi = 2;
+  optional int64 lo = 3;
+}
+
+message FileIdPB {
+  optional RowsetIdPB rowset_id = 1;
+  optional int32 segment_index = 2;
+}
+
+enum LogEntryTypePB {
+  INSERT_RANGE_PB = 0;
+  UPDATE_PB = 1;
+  DELETE_PB = 2;
+  EMPTY_PB = 3;
+}
+
+message InsertRangePB {
+  optional FileIdPB file_id = 1;
+  optional int32 start_row_id = 2;
+  optional int32 num_rows = 3;
+}
+
+message UpdatePB {
+  optional FileIdPB before_file_id = 1;
+  optional int32 before_row_id = 2;
+  optional FileIdPB after_file_id = 3;
+  optional int32 after_row_id = 4;
+}
+
+message DeletePB {
+  optional FileIdPB file_id = 1;
+  optional int32 row_id = 2;
+}
+
+message LogEntryPB {
+  optional LogEntryTypePB entry_type = 1;
+  optional InsertRangePB insert_range_data = 2;
+  optional UpdatePB update_data = 3;
+  optional DeletePB delete_data = 4;
+}
+
+message PageContentPB {
+  repeated LogEntryPB entries = 1;
+}
+
+enum PageType {
```

Comment:
add comments about the layout of binlog file, e.g. headers/pages/legentries

Reply:
Add file layout description on `BinlogFileWriter`